### PR TITLE
bwrap: mount readonly nix store

### PR DESCRIPTION
**Pull request by malt3**
#2201 breaks mkosi on NixOS, since binaries are located in `/nix/store` (and not in `/usr`).
The first commit simply mounts the nix store, if available.
The second commit was required to get ssl certificates to work on NixOS (without using a tools tree), since dnf would fail with certificate warnings.
On nixos, certificates are stored as follows:

- `/etc/ssl/certs/ca-bundle.crt` symlink to `/etc/static/ssl/certs/ca-bundle.crt`
- `/etc/static` symlink to `/nix/store/<HASH>-etc/etc`
I'm not sure if the second commit is mergeable as-is (I think it might break the tools tree method), but it would be amazing if we could find a working solution.

### CODE REVIEW

File: mkosi/bubblewrap.py [hunk 1]
```diff
@@ -80,6 +80,8 @@ def bwrap(
         "bwrap",
         "--ro-bind", "/usr", "/usr",
         "--ro-bind-try", "/nix/store", "/nix/store",
+        "--ro-bind-try", "/etc/ssl", "/etc/ssl",
```
Comment (DaanDeMeyer) [inline mkosi/bubblewrap.py:83]: We already handle `/etc/ssl` in `finalize_mounts()`.

File: mkosi/bubblewrap.py [hunk 2]
```diff
@@ -80,6 +80,8 @@ def bwrap(
...
+        "--ro-bind-try", "/etc/static", "/etc/static",
```
Comment (DaanDeMeyer) [inline mkosi/bubblewrap.py:84]: Can you move this to `finalize_mounts()`?

File: mkosi/bubblewrap.py (hunk 1 above)
Comment (malt3) [reply mkosi/bubblewrap.py:83]: Oh. I misread the code in `finalize_mounts`. Will fix this.

**Review by DaanDeMeyer — APPROVED** [review]

**Comment by DaanDeMeyer** [comment]
Actually, if /etc/static is a symlink, we shouldn't bind mount it. @malt3 Can you switch /etc/static to use --symlink? There's already some code in bwrap() that uses --symlink, you can put it below that.

**Review by DaanDeMeyer — APPROVED** [review]

**Review by behrmann — APPROVED** [review]

**Pull request merged**